tests/util/test_full_block_utils.py

```diff
@@ -129,20 +131,21 @@ def get_foliage_transaction_block():
     )
 
 
-def get_transactions_info(height):
-    yield None
+def get_transactions_info(height, foliage_transaction_block):
     farmer_coin, pool_coin = rewards(uint32(height))
     reward_claims_incorporated = [farmer_coin, pool_coin]
     fees = uint64(random.randint(0, 150000))
-
-    yield TransactionsInfo(
-        hsh(),  # generator_root
-        hsh(),  # generator_refs_root
-        g2(),  # aggregated_signature
-        fees,
-        uint64(random.randint(0, 12000000000)),  # cost
-        reward_claims_incorporated,
-    )
+    if foliage_transaction_block:
+        yield TransactionsInfo(
+            hsh(),  # generator_root
+            hsh(),  # generator_refs_root
+            g2(),  # aggregated_signature
+            fees,
+            uint64(random.randint(0, 12000000000)),  # cost
+            reward_claims_incorporated,
+        )
+    else:
+        yield None
```

Comment:
you should check this condition and return `None` as early as possible in this function, to avoid paying the cost of generating all those other values. Alternatively you could move those into the if-block as well. But an early exit saves on indentation levels